poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd

```diff
@@ -0,0 +1,166 @@
+library(arrow)
+library(dplyr)
+library(magrittr)
+library(stringr)
+library(ggplot2)
+library(sf)
+library(lubridate)
+source("useful_spatial_functions_data_extraction.R")
+```
+
+# Connecting to RIMReP collection
+
+Connecting to the AIMS Sea Surface Temperature Monitoring Program. This can take a minute or so.
+
+```{r connect}
+#Establishing connection
+data_bucket <- s3_bucket("s3://rimrep-data-public/091-aims-sst/test-50-64-spatialpart")
+
+#Accessing dataset
+data_df <- open_dataset(data_bucket)
+```
+
+# Extracting monitoring sites and their coordinates
+
+```{r sites_bin}
+sites_shp <- data_df %>% 
+  #We select unique sites included in the dataset
+  distinct(site, subsite, lon, lat) %>%
+  #This will load them into memory
+  collect()
+
+#Creating shapefile of unique locations
+sites_shp <- sites_shp %>% 
+  #Creating column to identify deployment location
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA)) %>%
+  #Turning into shapefile
+  st_as_sf(coords = c("lon", "lat"), crs = 4326)
+
+#Checking results
+glimpse(sites_shp)
+```
+# Identifying AIMS sites within GBR feature of interest
+In this section, we will use the functions included in the [useful_spatial_functions_data_extraction](https://github.com/aodn/rimrep-examples/blob/main/poc-data-api/useful_spatial_functions_data_extraction.R) script to load GBR features of interest. We will then use the boundaries of the feature of interest to identify the names of the sites within these limits.
+
+We will use the `gbr_features` function to access the boundaries for our sites of interest. If no parameters are given to this function, it will load all features recognised by the GBRMPA. You could also provide the name of the site using the `site_name` parameter or the GBRMPA unique ID using the `site_id` parameter. You can also use both parameters at the same time in case you have the names for some sites and the IDs for others.
+
+As an example, we will use **U/N Reef** or unnamed reefs. We chose them simply because they cover a large area and will overlap with the most number of AIMS monitoring sites.
+
+```{r gbr_features}
+un_reefs <- gbr_features(site_name = "u/n reef")
+
+#Checking results
+un_reefs
+```
+## Identifying sites of interest
+Depending on the amount of polygons used to identify the sites of interest, this may take a couple of minutes.  
+  
+We will plot the original AIMS monitoring sites (black), the polygon of interest (pink) and the sites within polygons (blue) to check that sites have been correctly identified.
+
+Note that unless the AIMS site is within a polygon, it will not to be identified in our final list.
+
+```{r sites_int}
+site_list <- sites_of_interest(sites_shp, un_reefs)
+
+#Plotting results

```

Comment:
How about plotting using Leafleat instead, so the user can zoom in/out the map

Reply:
I added a `leaflet` map to the notebook. This way users can see how to produce an interactive and static map. We can remove the static version using `ggplot2` if you think is best